projects/ngrx.io/content/guide/store/configuration/runtime-checks.md

```diff
@@ -50,10 +50,10 @@ To fix the above violation, a new reference to the state has to be created:
 
 ```ts
 export const reducer = createReducer(initialState,
-  on(addTodo, state => ({
+  on(addTodo, (state, { todo } => ({
```

Comment:
```suggestion
  on(addTodo, (state, { todo }) => ({
```

Reply:
Could you also take another look at the other examples on this page, because most of them are wrong.
Where `payload` is used we need to change it, add the action as a second parameter and destruct it to have the right property. 

```ts
export const reducer = createReducer(initialState,
  on(addTodo, state => ({
    // Violation 1: we assign a new value to `todoInput` directly
    state.todoInput = '',
    // Violation 2: `push` modifies the array
    state.todos.push(action.payload)
  }))
);
```

should be

```ts
export const reducer = createReducer(initialState,
  on(addTodo, (state, { todo }) => ({
    // Violation 1: we assign a new value to `todoInput` directly
    state.todoInput = '',
    // Violation 2: `push` modifies the array
    state.todos.push(todo)
  }))
);
```